lib/services/messenger/translators/discourse.ts

```diff
@@ -240,19 +240,29 @@ export class DiscourseTranslator extends TranslatorScaffold implements Translato
 	 * @returns         Promise that resolves to emit instructions.
 	 */
 	private static convertReadConnectionResponse(
-		message: TransmitInformation, response: DiscourseResponse
+		message: TransmitInformation, response: DiscoursePostSearchResponse
 	): Promise<IdentifyThreadResponse> {
-		const idFinder = new RegExp(`${message.source.service} thread ([\\w\\d-+\\/=]+)`, 'i');
-		if (response.posts.length > 0) {
-			return Promise.resolve({
+		const idFinder = new RegExp(`${message.source.service} thread ([\\w\\d-+\\/=\\_]+)`, 'i');
+		// Maps each item to its matched ID, and then removes falsey values.
+		const matches = _.compact(_.map(
+			response.posts,
+			(comment) => {
 				// Because Discourse assumes that when my POST contains:
 				// hyphen hyphen ("--") then what I mean is en dash ("–")
 				// hyphen hyphen hyphen ("---") then what I mean in em dash ("—")
-				thread: response.posts[0].blurb.replace('–', '--').replace('—', '---').match(idFinder)[1],
+				const fixedText = comment.blurb.replace(/–/g, '--').replace(/—/g, '---');
```

Comment:
Just so I understand this, the blurb is the post text, right? So what if someone's written:
```
var myVar = 1 - 2;
```
?

If not and this is purely private preamble, obviously it's not a problem.

Reply:
@hedss are you making this comment because all the dashes look the same in a fixed width font?   There are actually three different dash characters in that line.

Reply:
Same question applied, if someone uses an endash or emdash in a comment, will it get replaced?

Reply:
Yes, it will, however...
1) This replaced text is only used to search for IDs, which have been witnessed containing double hyphens, but not witnessed with en or em dashes.
2) Discourse does this replacement in a destructive, non-reversible manner.